meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register

```diff
@@ -100,31 +88,20 @@ class Pkcs11:
             cls._initialize()
         return cls()
 
-
-class SssCli:
-    @contextmanager
-    def i2c_session(*args, **kwds):
-        log.info("Connecting to ssscli...")
-        subprocess.check_call(["ssscli", "connect", "se05x", "t1oi2c", "none"])
-        try:
-            yield
-        finally:
-            log.info("Disconnecting from ssscli")
-            subprocess.check_call(["ssscli", "disconnect"])
-
+class FioCli:
     @classmethod
-    def read_id_list(cls) -> List[str]:
-        with cls.i2c_session():
-            out = subprocess.check_output(["ssscli", "se05x", "readidlist"])
-            lines = out.decode().splitlines()
-            # lines are like: Key-Id: 0Xf0000003   BINARY .....
-            return [x.split(" ")[1] for x in lines if x]
+    def read_id_list(self) -> List[str]:
+        command = ["fio-se05x-cli", "--list-objects", "all"]
+        process = subprocess.Popen(command, stdout=None, stderr=subprocess.PIPE)
+        stdout,stderr = process.communicate()
+        lines = stderr.decode().splitlines()
+        pattern = r'Key-Id:\s+(0x[0-9a-fA-F]+)'
```

Comment:
you are ignoring stdout and parsing stderr instead? Generally if you are ignoring a value in python you'd do:
```
_, stderr = process.communicate()
```

Also - python has a tool called "black". You can run it on this to auto-format it in canonical python3 style

Reply:
> you are ignoring stdout and parsing stderr instead? Generally if you are ignoring a value in python you'd do:
> 
> ```
> _, stderr = process.communicate()
> ```
> 
> Also - python has a tool called "black". You can run it on this to auto-format it in canonical python3 style

Wow, thanks @doanac Black is amazing! :)